extension-manifest-v3/src/utils/api.js

```diff
@@ -77,3 +83,45 @@ export async function session() {
 
   return jwtDecode(accessToken);
 }
+
+export async function getUserOptions() {
+  const userId = await getCookie('user_id');
+  const accessToken = await getCookie('access_token');
+  const csrfToken = await getCookie('csrf_token');
+
+  const data = await fetch(`${ACCOUNT_URL}/options/${userId}`, {
+    headers: {
+      'Content-Type': 'application/vnd.api+json',
+      Authorization: `Bearer ${accessToken}`,
+      'X-CSRF-Token': csrfToken,
+    },
+    credentials: 'omit',
+  });
+
+  return (await data.json()).data.attributes.options || {};
+}
+
+export async function setUserOptions(options) {
+  const userId = await getCookie('user_id');
+  const accessToken = await getCookie('access_token');
+  const csrfToken = await getCookie('csrf_token');
+
+  const data = await fetch(`${ACCOUNT_URL}/options/${userId}`, {
+    method: 'PATCH',
+    headers: {
+      'Content-Type': 'application/vnd.api+json',
+      Authorization: `Bearer ${accessToken}`,
+      'X-CSRF-Token': csrfToken,
+    },
+    credentials: 'omit',
+    body: JSON.stringify({
+      data: {
+        type: 'options',
+        id: userId,
+        attributes: { options },
+      },
+    }),
+  });
+
+  return (await data.json()).data.attributes.options || {};
```

Comment:
Looks like this can fail if the session is expired:

```
Error while syncing options:  TypeError: (intermediate value).data is undefined
```

If the session is expired, the fetch above will see a 401 with a JSON body like that:

```
{
  "errors": [
    {
      "title": "token is not valid",
      "status": "401",
      "code": "10020",
      "links": {
        "about": "https://github.com/ghostery/auth/wiki/Errors#10020--token is not valid"
      }
    }
  ]
}
```

Same in getUserOptions, I suppose. If there is a refresh token, the extension could try to renew the Bearer token and if successful, retry the get/set operation that failed.

Reply:
Hm.. but we do a refresh of the token, each time we call for `session()`, and it is used in sync process:

https://github.com/ghostery/ghostery-extension/pull/1234/files#diff-ab787eda050f8d097068bea78c9043e942c4480fdb99acf1f1c0213b0bbd859dR147

Reply:
We can add an additional check, or throw an error if the status is not 2xx, but the effect is the same, as the sync process is prepared to catch an error.

The sync requires a check of the user session, so I though that is redundant to make it again in those functions, as they are only used there.